src/main/java/carpet/script/CarpetEventServer.java

```diff
@@ -653,6 +654,25 @@ public void onWorldEventFlag(ServerWorld world, BlockPos pos, int flag)
                 );
             }
         };
+        public static final Event CARPET_RULE_CHANGES = new Event("carpet_rule_changes", 2, true)
+        {
+            @Override
+            public void onCarpetRuleChanges(ParsedRule<?> rule, ServerCommandSource source)
+            {
+                String identifier = rule.settingsManager.getIdentifier();
+                final String namespace;
+                if (!identifier.equals("carpet")) 
+                {
+                    namespace = identifier+":";
+                } else { namespace = "";}
+                handler.call(
+                        () -> Arrays.asList(
+                                new StringValue(namespace+rule.name),
+                                new StringValue(rule.getAsString())
```

Comment:
can technically fetch the proper 'value' (NumericValue for numbers etc), but I am not sure if you would want to merge more carpet (modded) code with scarpet (that 'pretends' is just a missing vanilla piece and assumes no mods exist)

Reply:
I decided to let it always a StringValue since you can convert those/check what they are easily with Scarpet functions. Also there would be some "fake types", like `COMMAND` rules usually being booleans, but can also be the `ops` String and the numbers `0-4` for perms, so it may be counter-intuitive to find a COMMAND `false` as a String but the `carpets` `false` as a boolean. Also the code is way simpler.

Reply:
Yeah, that makes sense - the client should know what value types to expect, and rules values can be more flexible in this regard than native scarpet values (bool+'ops'+1-4). Lets leave it strings